network/errors.go

```diff
@@ -0,0 +1,33 @@
+package network
+
+import "fmt"
+
+// NotSubscribedError is returned when the peer is not subscribed to a
+// specific topic.
+type NotSubscribedError struct {
+	TopicID TopicID
+}
+
+func (e NotSubscribedError) Error() string {
+	return fmt.Sprintf("Not subscribed to the '%s' topic", e.TopicID.String())
+}
+
+// InvalidTopicError is returned when the Pub-Sub topic is invalid.
+type InvalidTopicError struct {
+	TopicID TopicID
+}
+
+func (e InvalidTopicError) Error() string {
+	return fmt.Sprintf("invalid topic: %s",
```

Comment:
I think it's better to have same syntax for returning errors, at `Error() string` method for `NotSubscribedError` struct you've implemented `return fmt.Sprintf("Not subscribed to the '%s' topic", e.TopicID.String())` which `%s` is in `' '` but in here it isn't and it would be better to have same syntax for here as other implementations.

Reply:
Good point, Please check.